src/node_data.ts

```diff
@@ -114,13 +114,20 @@ function importSingleNode(node: Node, fallbackKey?: Key) {
   return data;
 }
 
+/**
+ * In lib.dom.d.ts in typescript v3.03 and before both Node.firstChild and Node.nextSibling were of type Node
+ * Starting in v3.1 firstChild is of type ChildNode, causing a type error.  Declaring type NodeOrChild as
+ * a union type allows the code to compile under both versions.
+ */
+type NodeOrChild = ChildNode | Node | null;
+
 /**
  * Imports node and its subtree, initializing caches.
  */
 function importNode(node: Node) {
```

Comment:
The Typescript compiler does not complain in either v3.0 or v3.1 because ChildNode in v3.1 extends Node so it has no problem with a function taking a Node as a parameter being given a ChildNode.

Perhaps it is cleaner to define a NodeOrNull type and then assert that child is a NodeOrNull.  Again, the compiler is OK with this even though in v3.1 node.firstChild is a FirstChild as that type extends the Node type.  In v3.0 there is no issue because all of these are Node types anyway.  I've checked this in v3.0 and v3.1 and it produces no errors so I'll amend the pull request.